src/shared/components/getting-started-section/index.js

```diff
@@ -1,30 +1,157 @@
-
+import {
+  LiveProvider,
+  LiveError
+} from 'react-live'
+import LiveEditor from 'shared/components/react-live/live-editor'
+import LivePreview from 'shared/components/react-live/live-preview'
 import HexSvg from 'shared/media/backgrounds/hexagons.svg'
-import StepsList from 'shared/components/getting-started-section/steps-list'
 import Button from 'shared/components/button'
+import { transformCode, log, getIpfs, preload } from 'utils/react-live-utils'
 import styles from './index.module.css'
+import SyntaxHighlighter from 'shared/components/syntax-highlighter'
+import 'prismjs/themes/prism-okaidia.css'
 
-const GettingStarted = ({ intl: { messages } }) =>
-  (
-    <div className={ styles.container }>
-      <div className={ styles.backgroundSvg }>
-        <div className={ styles.hex1 }><HexSvg /></div>
-        <div className={ styles.hex2 }><HexSvg /></div>
-      </div>
-      <div className={ styles.content }>
-        <h1>{ messages.gettingStarted.sectionTitle }</h1>
-        <ReactMarkdown className={ styles.sectionDescription } source={ messages.gettingStarted.sectionDesc } />
-        <div className={ styles.panel } >
-          <StepsList />
+const code = `const node = new IPFS()
+
+const data = 'Hello, <YOUR NAME HERE>'
+
+// once the node is ready
+node.once('ready', () => {
+  // convert your data to a Buffer and add it to IPFS
+  node.files.add(node.types.Buffer.from(data), (err, files) => {
+    if (err) return console.error(err)
+
+    // 'hash', known as CID, is a string uniquely addressing the data
+    // and can be used to get it again. 'files' is an array because
+    // 'add' supports multiple additions, but we only added one entry
+    console.log(files[0].hash)
+  })
+})`
+
+const codeGet = (cid) => `const node = new IPFS()
+
+node.once('ready', () => {
+  node.files.cat('${cid}', (err, data) => {
+    if (err) return console.error(err)
+
+    // convert Buffer back to string
+    console.log(data.toString())
+  })
+})`
+
+class GettingStarted extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {ipfsLoaded: false, outputAdd: null, outputGet: null, cid: null}
+    this.scopeAdd = null
+    this.scopeGet = null
+    this.refAdd = React.createRef()
+    this.refGet = React.createRef()
+    this.ipfs = null
+  }
+
+  handleChange = (editor) => (content) => {
+    if (content instanceof Error) {
+      content = content.message
+    }
+
+    if (typeof content !== 'string') {
+      return
+    }
+
+    if (editor === 'add') {
+      this.setState({outputAdd: content})
+    } else {
+      this.setState({outputGet: content})
+    }
+
+    if (content) {
+      if (typeof content === 'string' && content.length === 46 && content !== this.state.cid) {
+        preload(content)
+        this.setState({cid: content})
+      }
+    }
+  }
+  handleRunClick = (editor) => () => {
+    if (editor === 'add') {
+      this.refAdd.current.run()
+    } else {
+      this.refGet.current.run()
+    }
+  }
```

Comment:
Regarding the consistency of code in this project (and my personal preference 😛), could you put the handlers above after the `render` function?

Reply:
done